presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionInterpreter.java

```diff
@@ -882,7 +882,7 @@ protected Object visitFunctionCall(FunctionCall node, Object context)
             }
 
             // do not optimize non-deterministic functions
-            if (optimize && (!function.isDeterministic() || hasUnresolvedValue(argumentValues))) {
+            if (optimize && (!function.isDeterministic() || hasUnresolvedValue(argumentValues) || node.getName().toString().equals("grouping"))) {
```

Comment:
When creating the `FunctionCall` node I use a `LongLiteral` as a place holder for the `groupid` channel thus making all arguments to the function literals. Consequently, the `ExpressionInterpreter` attempts to compute and inline the function during planning. This can be prevented by (1) marking the function as non-deterministic or (2) using a `QualifiedNameReference` instead of the `LongLiteral` as the place holder for the `groupid` channel or (3) the code I added above.

(1) doesn't make sense because the function is deterministic, for the same inputs it will generate the same outputs. I tried (2) but ran into all sorts of trouble in `TranslationMap`. I think (3) is the most elegant solution.